modules/common/condition/funcs.go

```diff
@@ -0,0 +1,205 @@
+}
+
+// MarkTrue sets Status=True for the condition with the given type.
+func (conditions *Conditions) MarkTrue(t Type, messageFormat string, messageArgs ...interface{}) {
+	conditions.Set(TrueCondition(t, messageFormat, messageArgs...))
+}
+
+// MarkFalse sets Status=False for the condition with the given type.
+func (conditions *Conditions) MarkFalse(t Type, reason string, severity Severity, messageFormat string, messageArgs ...interface{}) {
+	conditions.Set(FalseCondition(t, reason, severity, messageFormat, messageArgs...))
+}
+
+// MarkUnknown sets Status=Unknown for the condition with the given type.
+func (conditions *Conditions) MarkUnknown(t Type, reason, messageFormat string, messageArgs ...interface{}) {
+	conditions.Set(UnknownCondition(t, reason, messageFormat, messageArgs...))
+}
+
+// IsTrue is true if the condition with the given type is True, otherwise it return false
+// if the condition is not True or if the condition does not exist (is nil).
+func (conditions *Conditions) IsTrue(t Type) bool {
+	if c := conditions.Get(t); c != nil {
+		return c.Status == corev1.ConditionTrue
+	}
+	return false
+}
+
+// IsFalse is true if the condition with the given type is False, otherwise it return false
+// if the condition is not False or if the condition does not exist (is nil).
+func (conditions *Conditions) IsFalse(t Type) bool {
+	if c := conditions.Get(t); c != nil {
+		return c.Status == corev1.ConditionFalse
+	}
+	return false
+}
+
+// IsUnknown is true if the condition with the given type is Unknown or if the condition
+// does not exist (is nil).
+func (conditions *Conditions) IsUnknown(t Type) bool {
+	if c := conditions.Get(t); c != nil {
+		return c.Status == corev1.ConditionUnknown
+	}
+	return true
+}
+
+// Sort - Sorts the so that the Ready condition always goes first, followed by all the other
+// conditions sorted by Type. This makes it easy to identify the overall state of
+// the service
+func (conditions *Conditions) Sort() {
+	// Sorts conditions for convenience of the consumer, i.e. cli client.
+	// According to this the Ready condition always goes first, followed by all the other
+	// conditions sorted by Type. This makes it easy to identify the overall state of
+	// the service
+	sort.Slice(*conditions, func(i, j int) bool {
+		return less(&(*conditions)[i], &(*conditions)[j])
+	})
+}
+
+// less returns true if a condition is less than another with regards to the
+// to order of conditions designed for better consumption i.e. cli client.
+// According to this the Ready condition always goes first, followed by all the other
+// conditions sorted by Type. This makes it easy to identify the overall state of
+// the service
+func less(i, j *Condition) bool {
+	return (i.Type == ReadyCondition || i.Type < j.Type) && j.Type != ReadyCondition
+}
+
+// hasSameState returns true if a condition has the same state of another
+func hasSameState(i, j *Condition) bool {
+	return i.Type == j.Type &&
+		i.Status == j.Status &&
```

Comment:
I'm not sure what are the rules of go indentation but this sames strange to have it over-indented with one column

Reply:
I need to get used to these strangeness of golang